loghero/LogHeroPluginSettings.php

```diff
@@ -2,19 +2,139 @@
 
 namespace LogHero\Wordpress;
 use \LogHero\Client\LogTransportType;
+use LogHero\Client\RedisOptions;
 
 
 class LogHeroPluginSettings {
+    public static $apiKeyOptionName = 'api_key';
+    public static $redisUrlOptionName = 'redis_url';
+    public static $redisKeyPrefixOptionName = 'redis_key_prefix';
+    public static $apiEndpointOptionName = 'api_endpoint';
+    public static $useSyncTransportOptionName = 'use_sync_transport';
+    public static $disableTransportOptionName = 'disable_transport';
+    public static $defaultRedisKeyPrefix = 'io.loghero:wp';
 
-    public static function getTransportType() {
-        $useSyncTransport = static::getOption('use_sync_transport');
+    private $settingsStorage;
+    private $hasWordPress;
+
+    private $apiKey;
+    private $transportType;
+    private $redisOptions;
+    private $apiEndpoint;
+
+    public function __construct($settingsStorage = null, $hasWordPress = null) {
+        if ($hasWordPress === null) {
```

Comment:
Why is there a check for word press and why is there no use of this flag (except for the tests)?

Reply:
Actually, this flag is only for testing to simulate the async environment (see my comment above).